addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -0,0 +1,275 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.evohome.internal.RunnableWithTimeout;
+import org.openhab.binding.evohome.internal.api.EvohomeApiClient;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Gateway;
+import org.openhab.binding.evohome.internal.api.models.v2.response.GatewayStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystem;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystemStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Zone;
+import org.openhab.binding.evohome.internal.api.models.v2.response.ZoneStatus;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides the bridge for this binding. Controls the authentication sequence.
+ * Manages the scheduler for getting updates from the API and updates the Things it contains.
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeAccountBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EvohomeAccountBridgeHandler.class);
+    private final HttpClient httpClient;
+    private EvohomeAccountConfiguration configuration;
+    private EvohomeApiClient apiClient;
+    private List<AccountStatusListener> listeners = new CopyOnWriteArrayList<AccountStatusListener>();
+
+    protected ScheduledFuture<?> refreshTask;
+
+    public EvohomeAccountBridgeHandler(Bridge thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(EvohomeAccountConfiguration.class);
+
+        if (checkConfig()) {
+            try {
+                apiClient = new EvohomeApiClient(configuration, this.httpClient);
+            } catch (Exception e) {
+                logger.error("Could not start API client", e);
+                updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Could not create evohome API client");
+            }
+
+            if (apiClient != null) {
+                // Initialization can take a while, so kick it off on a separate thread
```

Comment:
... and as it can take a while, you should set the thing status to UNKNOWN here, before leaving the initialize method.